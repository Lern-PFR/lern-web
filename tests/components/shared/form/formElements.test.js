import { shallow } from 'enzyme';

import {
	LabeledCheckbox,
} from 'components/shared/form';

describe('Exported form elements', () => {
	describe('LabeledCheckbox', () => {
		it('should match previous snapshot', () => {
			const sut = (<LabeledCheckbox id="lcb">Primary button</LabeledCheckbox>);
			const wrapper = shallow(sut);

			expect(wrapper).toMatchSnapshot();
		});

		it('should match previous snapshot for checked state', () => {
			const sut = (<LabeledCheckbox checked id="lcb">Primary button</LabeledCheckbox>);
			const wrapper = shallow(sut);

			expect(wrapper).toMatchSnapshot();
		});

		it('should match previous snapshot for disabled state', () => {
			const sut = (<LabeledCheckbox disabled id="lcb">Primary button</LabeledCheckbox>);
			const wrapper = shallow(sut);

			expect(wrapper).toMatchSnapshot();
		});

		it('should match previous snapshot for checked and disabled states', () => {
			const sut = (<LabeledCheckbox disabled checked id="lcb">Primary button</LabeledCheckbox>);
			const wrapper = shallow(sut);

			expect(wrapper).toMatchSnapshot();
		});
	});
});
