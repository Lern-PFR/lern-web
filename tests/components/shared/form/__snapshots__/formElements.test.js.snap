// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exported form elements LabeledCheckbox should match previous snapshot 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={false}
    disabled={false}
    id="lcb"
  />
  <LabelComponent
    color="initial"
    element="lcb"
  >
    Primary button
  </LabelComponent>
</styled.div>
`;

exports[`Exported form elements LabeledCheckbox should match previous snapshot for checked and disabled states 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={true}
    disabled={true}
    id="lcb"
  />
  <LabelComponent
    color="#707070"
    element="lcb"
  >
    Primary button
  </LabelComponent>
</styled.div>
`;

exports[`Exported form elements LabeledCheckbox should match previous snapshot for checked state 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={true}
    disabled={false}
    id="lcb"
  />
  <LabelComponent
    color="initial"
    element="lcb"
  >
    Primary button
  </LabelComponent>
</styled.div>
`;

exports[`Exported form elements LabeledCheckbox should match previous snapshot for disabled state 1`] = `
<styled.div
  alignItems="center"
  display="flex"
>
  <CheckboxComponent
    checked={false}
    disabled={true}
    id="lcb"
  />
  <LabelComponent
    color="#707070"
    element="lcb"
  >
    Primary button
  </LabelComponent>
</styled.div>
`;
